hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ResourceLocalizationService.java

```diff
@@ -212,6 +213,15 @@ public ResourceLocalizationService(Dispatcher dispatcher,
     this.delService = delService;
     this.dirsHandler = dirsHandler;
 
+    this.disablePrivateVis = UserGroupInformation.isSecurityEnabled() &&
+        context.getConf().getBoolean(
+            YarnConfiguration.NM_SECURE_MODE_USE_POOL_USER,
+            YarnConfiguration.DEFAULT_NM_SECURE_MODE_USE_POOL_USER);
+    if (this.disablePrivateVis) {
+      LOG.info("When " + YarnConfiguration.NM_SECURE_MODE_USE_POOL_USER +
```

Comment:
All resources can either be `public` or `application`. It seems this constraint is added since a `local-user` may get assigned to different `real-users`. Can you comment on the cases where a real user application may require `private` resources and `local-user-pooling` is enabled?

Reply:
Sure, will add comments. Basically if a real user application require private resources, when local user pooling is enabled, we will treat the private resources as application resources. This means for each application we will download (localize) the resources to application folder, and will delete it after the application is finished.